### DISCUSSION TITLE
[Help]: upgrading to v8 seems like it's missing a directory?

### DISCUSSION THREAD

**Discussion by jakeRPowell** [davidjerleke/embla-carousel#806] [Help] [answered]
### Summary

I upgraded from version 7 to version 8 in order to use the autoscroll plugin (this was the only way I managed to get it to work, so if there's a workaround I'd be happy with that). However now my jest tests break, i get a variety of errors but it seems that the issue is that it's looking for 

(node_modules/embla-carousel/src/components/OptionsHandler.ts:37:8)

but the src directory doesn't exist, have reinstalled a few times. Any ideas?

![Screenshot 2024-03-27 at 13 34 04](https://github.com/davidjerleke/embla-carousel/assets/43782301/2aa1610c-a211-4442-b5f0-6cb8b7186bf0)

I wasn't sure what info to add re error messages/my code, but happy to add anything anyone might think helpful 

Thanks in advance!

### If applicable, which variants of Embla Carousel are relevant to this question?

- [x] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [X] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by jakeRPowell** [comment]
just needed to install the core package :D

**Comment by jakeRPowell** [comment]
update! 

adding the core package and downgrading "embla-carousel-react" fixed (most of) the tests, but broke the carousel on the site, 

current configuration gives me nicely working components on the site, but the tests are broken

		"embla-carousel": "^8.0.0",
		"embla-carousel-auto-scroll": "^8.0.0",
		"embla-carousel-react": "^8.0.0",

**Reply by davidjerleke** [reply, marked as answer]
@jakeRPowell see:

- #516 
- For example, here’s a mock for [IntersectionObserver](https://github.com/davidjerleke/embla-carousel/blob/master/packages/embla-carousel/src/__tests__/mocks/intersectionObserver.mock.ts) you can use. 
- Here are [all mocks](https://github.com/davidjerleke/embla-carousel/tree/master/packages/embla-carousel/src/__tests__/mocks) Embla is using for its tests.

**Comment by jakeRPowell** [comment]
Thanks so much!
